core/src/main/java/com/radixdlt/rev2/modules/NodePersistenceModule.java

```diff
@@ -67,36 +67,62 @@
 import com.google.inject.AbstractModule;
 import com.google.inject.Provides;
 import com.google.inject.Singleton;
-import com.google.inject.multibindings.Multibinder;
 import com.radixdlt.consensus.safety.BerkeleySafetyStateStore;
 import com.radixdlt.consensus.safety.PersistentSafetyStateStore;
-import com.radixdlt.environment.NodeAutoCloseable;
+import com.radixdlt.consensus.safety.RocksSafetyStateStore;
+import com.radixdlt.environment.NodeRustEnvironment;
 import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.safety.RocksDbSafetyStore;
 import com.radixdlt.serialization.Serialization;
 import com.radixdlt.store.BerkeleyDbDefaults;
 import com.radixdlt.store.NodeStorageLocation;
 import com.radixdlt.utils.properties.RuntimeProperties;
 
-public class BerkeleySafetyStoreModule extends AbstractModule {
+public class NodePersistenceModule extends AbstractModule {
+  private final AddressBookPolicy policy;
+
+  public enum AddressBookPolicy {
+    ENABLED,
+    DISABLED
+  }
+
+  public NodePersistenceModule(AddressBookPolicy policy) {
+    this.policy = policy;
+  }
+
   @Override
   protected void configure() {
-    bind(PersistentSafetyStateStore.class).to(BerkeleySafetyStateStore.class);
-    Multibinder.newSetBinder(binder(), NodeAutoCloseable.class)
-        .addBinding()
-        .to(BerkeleySafetyStateStore.class);
+    if (policy == AddressBookPolicy.ENABLED) {
+      install(new AddressBookModule());
+    }
   }
 
   @Provides
   @Singleton
-  BerkeleySafetyStateStore safetyStateStore(
+  PersistentSafetyStateStore persistentSafetyStateStore(
       RuntimeProperties properties,
       Serialization serialization,
       Metrics metrics,
-      @NodeStorageLocation String nodeStorageLocation) {
-    return new BerkeleySafetyStateStore(
-        serialization,
-        metrics,
-        nodeStorageLocation,
-        BerkeleyDbDefaults.createDefaultEnvConfigFromProperties(properties));
+      @NodeStorageLocation String nodeStorageLocation,
+      NodeRustEnvironment environment) {
+
+    var berkeleySafetyStateStore =
```

Comment:
Can we move the creation of the `berkeleySafetyStateStore` inside the `!rocksSafetyStateStore.isMigrated()` check?

That way, we can avoid wasting time initializing Berkeley on start-up (and also I imagine avoid creating all the Berkeley files on disk) once we have migrated.

Reply:
Done